lib/turnout/ordered_options.rb

```diff
@@ -0,0 +1,62 @@
+module Turnout
+  class OrderedOptions < Hash
+    alias_method :_get, :[] # preserve the original #[] method
+    protected :_get # make it protected
+
+    def []=(key, value)
+      super(key.to_sym, value)
+    end
+
+    def [](key)
+      super(key.to_sym)
+    end
+
+    def method_missing(name, *args)
+      name_string = name.to_s
+      if name_string.chomp!('=')
+        self[name_string] = args.first
+      else
+        bangs = name_string.chomp!('!')
+
+        if bangs
+         value = fetch(name_string.to_sym)
+         raise(RuntimeError.new("#{name_string} is blank.")) if value.nil? || value.empty?
+         value
+        else
+          self[name_string]
+        end
+      end
+    end
+
+
+    def respond_to_missing?(name, include_private)
+      true
+    end
+  end
+
+
+  # +InheritableOptions+ provides a constructor to build an +OrderedOptions+
+  # hash inherited from another hash.
+  #
+  # Use this if you already have some hash and you want to create a new one based on it.
+  #
+  #   h = ActiveSupport::InheritableOptions.new({ girl: 'Mary', boy: 'John' })
+  #   h.girl # => 'Mary'
+  #   h.boy  # => 'John'
+  class InheritableOptions < Turnout::OrderedOptions
+    def initialize(parent = nil)
+      if parent.kind_of?(Turnout::OrderedOptions)
+        # use the faster _get when dealing with OrderedOptions
+        super() { |h,k| parent._get(k) }
+      elsif parent
+        super() { |h,k| parent[k] }
+      else
+        super()
+      end
+    end
+
+    def inheritable_copy
```

Comment:
This is used in the Turnout::Internationalization , method  ```i18n_config``` 
This is needed in case someone will use the configuration like this:
```ruby
Turnout.config.i18n = { some_key:  :value  } 
```
The code will transform the hash into a instance of OrderedOptions class to preserve compatibility with the existing code.